src/dune_rules/menhir.ml

```diff
@@ -50,6 +50,8 @@ end
 module Run (P : PARAMS) : sig end = struct
   open P
 
+  let cctx = Compilation_context.without_bin_annot cctx
```

Comment:
As it is, it wasn't immediately clear to me that this `cctx` was *only* used to compile the mock module. Perhaps the logic would be slightly clearer if this call was applied to the `cctx` which is passed to `ocamlc_i` further below instead of applying globally here. Just my 2c.

Reply:
Good point. I changed it.